CONTRIBUTING.md

```diff
@@ -3,11 +3,201 @@ Guidelines for Contributing
+- [ ] Is your code flexible enough to accommodate more use cases?
+- [ ] Read through your code again - is there any part that is not understandable?
+
+### Tests
+
+- [Autotester](http://wiki.dlang.org/Git_Commit_Tester) will automatically compile the code in your PR and test it on all supported platforms. For your first PR you need approval from any reviewer - ping them in case they forget.
+- [ ] Do all tests pass locally? You can run the tests of
+	- single module with `rdmd -main -unittest module.d`
+	- an entire package with `make -f posix.mak std/algorithm.test`
+	- all tests with `make -f posix.mak unittest` (add `-j NUMBER_OF_CORES` if you have multiple cores)
+	- for Windows have a look at [building Phobos under windows](http://wiki.dlang.org/Starting_as_a_Contributor#Windows_2))
+- [ ] Do your tests cover all cases? (you can check code coverage in the resulting `module.lst` of `rdmd -cov -main -unittest module.d`)
+
+### Review process
+
+- All PRs _must_ pass all tests for all platforms within [autotester](http://wiki.dlang.org/Git_Commit_Tester) before they will be merged
+- Big additions (new modules) should go to `std.experimental` first (after they pass the [review process](http://wiki.dlang.org/Review/Process))
+- Smaller additions like individual functions can be merged directly after [@andralex](https://github.com/andralex) approves
+- Refactoring or bug fixes just need approval (LGTM) from two reviewers and enough time passed (usually 2-3 days, depending on weekends and holidays) to give everyone an opportunity to shout
+- Trivial changes (e.g. small documentation enhancements) are usually merged in less than 72h - otherwise ping for reviewers
+- If your PR is stalled (no activity for 2-3 days) and you addressed all concerns, don't hesitate to ping your reviewers
+- Ping [@9il](https://github.com/9il) for anything math related
+- See [rebasing](#rebasing) if you need to sync your fork with master
+
+### Naming
+
+- [ ] Do you use clear variable names?
+- [ ] Does your PR have a clear and informative title?
+- [ ] Are your Git messages clear and informative (otherwise see `git rebase`)?
+- Avoid having many commits per PR (usually one commit per PR) - see [squashing](#squashing) to combine commits
+
+### Fixing issues
+
+- Bug fix PRs should have a separate commit for each bugzilla issue that it fixes
+- The message of the commit actually fixes the bug should start with "Fix issue X" where X is the number of the corresponding [bugzilla](https://issues.dlang.org) issue. For example:
+
+```
++ Refactor the code assist with the bug fix
++ Fix issue 17000 - map doesn't accept some range types
++ Fix issue 17001 - map doesn't work with some other range type
+```
+
+- Unrelated fixes, refactorings or additions should preferably go separate pull request(s) (tightly related fixes are okay in the same PR)
+
+Documentation
+-------------
+
+### Documentation style
+
```

Comment:
Maybe this is a good place to explain that the documentation is written in ddoc and to add a link to the ddoc [docs](http://dlang.org/spec/ddoc.html).